frontend/lib/uast/BuilderResult.cpp

```diff
@@ -338,6 +377,16 @@ bool BuilderResult::equals(const BuilderResult& other) const {
     }
   }
 
+  // Check that all the additional location maps match.
+  #define CHPL_LOCATION_MAP(ast__, location__) { \
+      auto& m1 = CHPL_ID_LOC_MAP(ast__, location__); \
+      auto& m2 = other.CHPL_ID_LOC_MAP(ast__, location__); \
+      if (m1.size() != m2.size()) return false; \
```

Comment:
note that the 'non-specialized' map code doesn't perform this check. Also note that this check is already performed by `operator==` for DenseMap:

```C++
template <typename DerivedT, typename KeyT, typename ValueT, typename KeyInfoT,
          typename BucketT>
bool operator==(
    const DenseMapBase<DerivedT, KeyT, ValueT, KeyInfoT, BucketT> &LHS,
    const DenseMapBase<DerivedT, KeyT, ValueT, KeyInfoT, BucketT> &RHS) {
  if (LHS.size() != RHS.size())
    return false;
```

Reply:
That's true, but I didn't want to make assumptions about the map code.

Reply:
I think that any reasonable implementation will do that check and it just gets in the way here. But your call whether to keep it.